homeassistant/components/roborock/scene.py

```diff
@@ -0,0 +1,75 @@
+"""Support for Roborock scene."""
+
+from __future__ import annotations
+
+import asyncio
+from typing import Any
+
+from roborock.containers import HomeDataScene
+
+from homeassistant.components.scene import Scene as SceneEntity
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import EntityDescription
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import RoborockConfigEntry
+from .coordinator import RoborockDataUpdateCoordinator
+from .entity import RoborockEntity
+
+
+async def _setup_coordinator(
+    coordinator: RoborockDataUpdateCoordinator,
+) -> list[SceneEntity]:
+    return [
+        RoborockSceneEntity(coordinator, scene)
+        for scene in await coordinator.get_scenes()
+    ]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: RoborockConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up scene platform."""
+    coordinator_entities = await asyncio.gather(
+        *[
+            _setup_coordinator(coordinator)
+            for coordinator in config_entry.runtime_data.v1
+        ],
+        return_exceptions=True,
+    )
+    async_add_entities(
+        entity
+        for entities in coordinator_entities
+        if isinstance(entities, list)
```

Comment:
We before discussed the principles behind exception handling and not silently ignoring exceptions, and this is again silently ignoring exceptions.

Reply:
handled